### DISCUSSION TITLE
latex  package difficulty

### DISCUSSION THREAD

**Discussion by glfan2814** [xournalpp/xournalpp#5600] [Q&A] [answered]
I'm not sure how to add packages to a template. while the default template works when I try to make a custom template and use that the test configuration just crashes. using Windows 11 and Miketex. I could not find any info. below is the code copied from the attempted template

`% This template uses the scontents package, which is only available on relatively recent TeX distributions. In case it is not available on your system, use the legacy_template.tex
\documentclass[varwidth=true, crop, border=5pt]{standalone}

% Packages
\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{chemfig}
\usepackage{longdivision}

% for storing in memory verbatim content to be reused later
\usepackage{scontents}

% Blank formula checking
\usepackage{ifthen}
\newlength{\pheight}

% Color support
\usepackage{xcolor}
\definecolor{xpp_font_color}{HTML}{%%XPP_TEXT_COLOR%%}
% User input
\begin{scontents}[store-env=preview]
	\( 
	\displaystyle
    %%XPP_TOOL_INPUT%%
	\)
\end{scontents}

\begin{document}
  % Check if the formula is empty
  \settoheight{\pheight}{\getstored[1]{preview}}%
  \ifthenelse{\pheight=0}{\GenericError{}{xournalpp:blankformula}{}{}}

  % Render the user input
  \textcolor{xpp_font_color}{\getstored[1]{preview}}
\end{document}`

**Comment by rolandlo** [comment, marked as answer]
The template looks correct. You have added the `chemfig` and `longdivision` packages to the default template. Make sure they are installed in your LaTeX installation and the MikTeX inputs database is refreshed. See e.g. [this link](https://sites.allegheny.edu/it/tutorials/latex-for-your-personal-computer/adding-more-latex-packages-to-your-miktex-installation/) for help on installing MikTeX packages.

By the way depending on the version of Xournal++ you use, you have a "Command Output" tab in the LaTeX tool that would show you the error message. 
![grafik](https://github.com/xournalpp/xournalpp/assets/40485433/99f45277-c3d6-4398-9f5f-fcacc030c311)

**Reply by glfan2814** [reply]
thanks. I was using the tex packages incorrectly.
